### PR TITLE
Compound: add `BigIcon`, `BigCheckmark` and `PageTitle` components

### PR DESCRIPTION
**Pull request by jmartinesp**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [x] Other : add compound components

## Content

Adds 3 new Compound components that will be needed for future development:
- BigIcon.
- BigCheckmark.
- PageTitle.

## Motivation and context

These components will be used in new designs and will replace older ones we were using.

## Screenshots / GIFs

Being recorded.

## Tests

Nothing, but you can check the screenshots look like the designs [in Figma](https://www.figma.com/file/G1xy0HDZKJf5TCRFmKb5d5/Compound-Android-Components?type=design&node-id=1960%3A540&mode=design&t=ngp1ciOA7zb9tNd6-1).

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [x] UI change has been tested on both light and dark themes
- [x] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/BigIcon.kt [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+     * @param modifier the modifier to apply to this layout
+     */
+    @Composable
+    operator fun invoke(
+        style: Style,
+        modifier: Modifier = Modifier,
+    ) {
+        val backgroundColor = when (style) {
+            is Style.Default -> ElementTheme.colors.bigIconDefaultBackgroundColor
+            Style.AlertSolid, Style.SuccessSolid -> ElementTheme.colors.bgCanvasDefault
```
Comment (bmarty) [inline libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/BigIcon.kt:96]: I am wondering if using a transparent color should be preferable in this case. I let you decide.

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/BigIcon.kt [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                BigIcon(style = style)
+            }
+        }
+    }
+}
+
+internal class BigIconStylePreviewProvider : PreviewParameterProvider<BigIcon.Style> {
+    override val values: Sequence<BigIcon.Style>
+        get() = sequenceOf(
+            BigIcon.Style.Default(Icons.Filled.CatchingPokemon),
```
Comment (bmarty) [inline libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/BigIcon.kt:148]: :)

**Review by bmarty — APPROVED** [review]
LGTM, thanks!

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/BigIcon.kt (hunk 1 above)
Comment (jmartinesp) [reply libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/BigIcon.kt:96]: Transparent is probably better, although it kind of contradicts its name. I'm asking in the Compound room to clarify this.

Comment (jmartinesp) [reply libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/BigIcon.kt:96]: I just confirmed it with them, it's the other way around: the solid ones should have a solid background (duh), the others should be transparent. I'll change those.

**Pull request merged**